pkg/auth/api/user/user_actions_test.go

```diff
@@ -0,0 +1,81 @@
+			expectsErr: false,
+		},
+		{
+			name:       "Large input exact match",
+			username:   "\"]`3Gy2*g6D8cX6],)PF[MJ3k(V;t:+q5!Man#_94`p_2be}TB]}}d6^L+Q@{N8[w.RwD<ZHDCBtc$eA\\\\747f`bA,PS5Vs%8TSf_mvGzKEXd!?Hxs2Z)N#dkE_}2K.zN]JWt>Q/b7rF6zSkXZ$w\\\"QN/Mg,tV4J~?Mj__srfGD%]q*K+uqD-vD[BxPw+&pt_w)4PVf:Ad#(}9(_>bkA3WdY-z'~?f2e.CA&U<rvU457XNT^UK_\\\"aaGr}=\\\"d95rxr9N\"",
+			password:   "\"]`3Gy2*g6D8cX6],)PF[MJ3k(V;t:+q5!Man#_94`p_2be}TB]}}d6^L+Q@{N8[w.RwD<ZHDCBtc$eA\\\\747f`bA,PS5Vs%8TSf_mvGzKEXd!?Hxs2Z)N#dkE_}2K.zN]JWt>Q/b7rF6zSkXZ$w\\\"QN/Mg,tV4J~?Mj__srfGD%]q*K+uqD-vD[BxPw+&pt_w)4PVf:Ad#(}9(_>bkA3WdY-z'~?f2e.CA&U<rvU457XNT^UK_\\\"aaGr}=\\\"d95rxr9N\"",
+			expectsErr: true,
+		},
+		{
+			name:       "Very Large input no match",
+			username:   "administrator",
+			password:   "k&M;HX`#ax$qc/}}.^R_3s$m8,m5>4$+x]c>Jz~zZ;``9VM_suT%w~TFc!!ZFMEn4]:t't/fu;vw!DfDfC^#HhQ&{HHM\\<MJ<{*x\\AZ{;\"@x9J`P'v5Htn)?}sw3h)XS{!2)ggkcw^4.k)p^hM${AMKt&`eeu{G?.w7z%<QF3'n7A8X;'D#h%4mE-9h;e63ZGxRCn9J7`>z9_wCS,q$!C.,S@3NT}$(\"^`v$xRBFrbSAj]J<DZt(y_dLW72{wNyreRs9)NMTM'V<Vm8:?.-{h-J@=xC?m#^GkMm;F]p77S*my!txc'LfXnHn^L'U%J+(`Gn*^#'8)E[VF\"}Tk6,[bc^N3%^[}zHt\"R.&+`?RTKf.}!74/rG[bBHuBqt)a$9pS!Vte>aK)CE)!naPjZ~)84k:\\GugL<w4<=v`dg'YC9C8UR;Va}9=uCE7.WW#9^DjH)U8(%9K~h%7.kgKe;Y/UB:s[<gCFL*]d7C]*wqNa?!7R8[A#XH2t*7]{y/N%@/.BpSWc4!x$DT6n/XX;bx,Bx:&]T,Y}\\^,E!@Bs_qB3rT<~39>'QNN\\QT:yRs:'f.{dh)=%;vNK(&HvDGT8X%+QxcdKYXz5zZ't2Tq$'@7<)Uh%Kc[z!j~9a_W+=}Yt,^{8td&Rrx~v#9S/TQ(-%@A(F6=fu$q$P4%$hvW%3=P_w>7d+phKFmBe'uzA<GhM>Y3M)hg\\/m=JtFHv9KBK)CD[w:MS\\-C4a)2`?\"nbvr[:Jp/dj.G+w$ucKB{y'LPjEsj>REWF/Gw~Au=A,j\\4Z8Mmw#}-nR(Av\"YekW7%<PRB+4T3]stA;5}~!sX+NQFc/#pDzy+E>jfT8%]tuseXB?[sQ%;[{,pTy`u7&:WGgf\"Lg>+HZ8]p%[]DVSbqXpUE]k.[kKp<X^Tsx22d$[.$m>>w==%26j;K\\*eMwH<3yc,w`d&]y7hh{VpMDJNxK#+-6TvKba;^E6gd'HGD!<2SfG8>)uV83wzc)s_j!agMPVq5'$;8q_~H;s.;[?N($r.s#!Q}>8Q:)cezSZwV+-f2&VN{R:nbe3,#8mJ=vJ-r'~g_b<X<Fjj`e(NEaaW93\"ML5/5_&=\\]}xYgg7x%z8NSk\"W}m%-CnkcBn(L7x499Am@93M2y6Y864;Wsm~9X9M6V!4mzA\\#8Z_wuA;B[H@`>+`-B'k!G2}u/WA)!4Fu_)E-fF\\Gt'z\\Q7uys@htugjrwQN5Jb6}rZS3&M;W#8J$.WUrHq+D<BE4S=3aR'p}FCpvUydV(*[4GgGvh;h(K&']zytSDvPhjQ#&[LvCwg>N)(~cq2,u;PW@'M;mQthb\\!W)W]'\"<vtT{8ABc<CM.!b$*zWA3+fQCEh)Z3Yc?z)qWvQdAU;))@zN+`gSPb<Pxnnh/&wA/B~q2')8a4,`=~]`WwHQ<TS5v5ehqhX^fctr~~j.me(+Av'jU<YKPEzkyG;gZ:$ApRTyLVn8x^h]C=-`>H#]u89ujXjZ~mf6T7Kw#z!PSuQE.jS@k`%8Vq!U`DS'9F.D:,WC]~:wDH-gmy%<A4*]vS\\&LJ[QgfG,`S;pz92tAv$Dkr}tW~Kr@<f[jrrUk/<TvEm$xRT5:_Wg&:/LLd$s/KfabHr6?@[<p\"bx!&%#(h\\j3*+Fuu5jLMn=C&&4~$g:3Su.mR%^\"G~\\gpaUDc)s(Ff;jP<';j?tm?2r_Pv~s~?y>,TNZa42?vC;MACq*\"r[8`!SK>][>E~DQ!d&J-{yE>x(v^x<3;['#N:%q-m9MJa]Kh;*y/]T?(;y9`NVsAC\"RL]Zn}uGcQup}t&^sU_kfb*n=gxd+V%&chJQLGqZ`!*qn&7KuKb[E6Zm6W/PkbB7S7q]}mytFH4dLu_PYA7;L)~_Udm.M/,^BuK`n,e+[*jT}@Xz=6~L.7]Y;4SV2\"wP)(b$4%U/uJL9k6GJp3.W^B2J6FYq/{EE<D>*WU5`48aXkav\\?/wWj~+X>8:X;2g\\b'F`\\P'h=pP~y>CdqM5u\"wRc(FvdSx\\E*:[s#,*8y~Ur;Kb?(_\\4H<RJts?@H)fGT?$$7.e/+\"g)N/Cw>a\".",
```

Comment:
I think these 2 "very large" cases are certainly overkill - we already cover really well with "large" cases as these cases don't cover any different execution paths.

Reply:
I'm just going to replace these with a generate function copied from here:
https://github.com/rancher/rancher/blob/eff74fb877091c2d1651f6f8acaf831235e0a464/pkg/data/management/resetpassword.go#L86-L100 
Normally I'm not too fond of nor deterministic test data, but I think it's fine in this case, we can just update the generate function if we want to ensure different characters are present in the future, or make a testdata folder for really long passwords if determinism becomes important. Also, do we have an official upper bound on passwords? I've tested with a max of 2048 characters which is what the very large input is, but I can cap it at whatever the official supported password length is if it exists.